msal/src/main/java/com/microsoft/identity/client/AuthenticationResult.java

```diff
@@ -39,30 +39,28 @@
 public final class AuthenticationResult implements IAuthenticationResult {
     //Fields for Legacy Cache
     private final String mTenantId;
-    private final String mRawIdToken;
     private final String mUniqueId;
 
     private final AccessTokenRecord mAccessToken;
-    private final IAccount mAccount;
-
-    public AuthenticationResult(@NonNull final ICacheRecord cacheRecord) {
-        mAccessToken = cacheRecord.getAccessToken();
-        mTenantId = cacheRecord.getAccount().getRealm();
-        mUniqueId = cacheRecord.getAccount().getHomeAccountId();
-        mRawIdToken = cacheRecord.getIdToken().getSecret();
-        mAccount = AccountAdapter.adapt(cacheRecord.getAccount());
+    private final com.microsoft.identity.client.tenantprofile.IAccount mAccount;
+
+    public AuthenticationResult(@NonNull final List<ICacheRecord> cacheRecords) {
```

Comment:
nit: is it possible that this could be an empty list?

Reply:
Not presently --  we only create this object in the event of success. Failure cases fall into the `MsalException` path where this object never gets constructed